globus_cli/commands/task/list.py

```diff
@@ -6,16 +6,141 @@
 from globus_cli.services.transfer import print_json_from_iterator, get_client
 
 
+class ISOTimeType(click.ParamType):
+    """
+    Enforces time inputs to be in either YYYY-MM-DD or YYYY-MM-DD HH:MM:SS
+    And that each value is a valid int that falls in the correct range
+    """
+
+    name = "ISO TIME"
+
+    def convert(self, value, param, ctx):
+        try:
+            if len(value) not in [10, 19]:
+                raise ValueError
+
+            year = int(value[:4])
+            if year <= 0:
+                self.fail("Invalid year {}".format(year))
+            month = int(value[5:7])
+            if month not in range(1, 13):
+                self.fail("Month {} must be in range 1-12.".format(month))
+            day = int(value[8:10])
+            if day not in range(1, 32):
+                self.fail("Day {} must be in range 1-31.".format(day))
+
+            if len(value) == 19:
+                hour = int(value[11:13])
+                if hour not in range(24):
+                    self.fail("Hour {} must be in range 0-23.".format(hour))
+                minute = int(value[14:16])
+                if minute not in range(60):
+                    self.fail(
+                        "Minute {} must be in range 0-59.".format(minute))
+                second = int(value[17:19])
+                if second not in range(61):
+                    self.fail(
+                        "Second {} must be in range 0-60.".format(second))
```

Comment:
I don't object to us doing our own parsing logic if there's a reason, but why not use [`datetime.strptime`](https://docs.python.org/2/library/datetime.html#datetime.datetime.strptime) or [`time.strptime`](https://docs.python.org/2/library/time.html#time.strptime)?

If we have a reason that we can't use those (e.g. the tighter coupling with `click`'s `fail()`), that should be stated in a comment or docstring.

If the error messaging is the only reason, I'd contest the necessity of custom parsing. A message like "20160124 is not of the format YYYY-MM-DD or of the format YYYY-MM-DD HH:MM:SS" seems fine.